### PR TITLE
[build-utils] Add CloudEventTrigger support for Lambda functions

### PR DESCRIPTION
**Pull request by Timer**
### TL;DR

Add support for CloudEvents triggers in Lambda functions.

### What changed?

- Added a new `CloudEventTrigger` interface in `types.ts` that defines how Lambda functions can receive CloudEvents via HTTP endpoints
- Extended the `Lambda` class to support an `experimentalTriggers` property that accepts an array of CloudEventTrigger definitions
- Added validation logic to ensure CloudEventTrigger objects are properly formatted
- Added comprehensive tests for the new CloudEventTrigger functionality
- Exported the CloudEventTrigger type for external use

### How to test?

Create a Lambda function with experimental CloudEvent triggers:

```typescript
const lambda = new Lambda({
  files,
  handler: 'index.handler',
  runtime: 'nodejs18.x',
  experimentalTriggers: [{
    triggerVersion: 1,
    specversion: '1.0',
    type: 'v1.test.vercel.com',
    httpBinding: {
      mode: 'structured',
      method: 'POST',
      pathname: '/webhooks/test'
    }
  }]
});
```

Run the test suite to verify validation logic and edge cases.

### Why make this change?

This change enables Lambda functions to define what types of CloudEvents they can handle as HTTP endpoints. This is particularly useful for event-driven architectures where functions need to respond to specific event types. The implementation follows the CloudEvents specification and supports HTTP protocol binding in structured mode, making it compatible with standard CloudEvents producers.

### CODE REVIEW

**Comment by Timer** [comment]
* **#13513** <a href="https://app.graphite.dev/github/pr/vercel/vercel/13513?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#13512** <a href="https://app.graphite.dev/github/pr/vercel/vercel/13512?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#13511** <a href="https://app.graphite.dev/github/pr/vercel/vercel/13511?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#13510** <a href="https://app.graphite.dev/github/pr/vercel/vercel/13510?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#13503** <a href="https://app.graphite.dev/github/pr/vercel/vercel/13503?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/vercel/vercel/13503?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: packages/build-utils/src/types.ts [hunk 1]
```diff
@@ -591,3 +591,69 @@ export interface Chain {
...
+ * Handles "com.vercel.queue.v1" events with queue-specific configuration.
+ */
+export interface CloudEventQueueTrigger
+  extends CloudEventTriggerBase<'com.vercel.queue.v1'> {
+  /**
+   * Queue configuration for this trigger (REQUIRED)
+   */
+  queue: {
+    /** Name of the queue subject to consume from (REQUIRED) */
+    subject: string;
```
Comment (ctgowrie) [inline packages/build-utils/src/types.ts]: topic?

**Review by ctgowrie — DISMISSED** [review]

File: packages/build-utils/src/lambda.ts [hunk 2]
```diff
@@ -29,6 +52,31 @@ export interface LambdaOptionsBase {
...
   framework?: FunctionFramework;
+  /**
+   * Whether this Lambda is private and requires special access controls.
+   * Must be set to true when using experimentalTriggers.
+   * Cannot be true without experimentalTriggers defined.
+   *
+   * @default false
+   * @experimental This feature is experimental and may change.
+   */
+  experimentalPrivate?: boolean;
```
Comment (ctgowrie) [inline packages/build-utils/src/lambda.ts]: I think we can imply this based on experimentTriggers being present or not before we add another field

**Review by ijjk — APPROVED** [review]
